staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -284,7 +313,8 @@ func (le *LeaderElector) tryAcquireOrRenew() bool {
 		le.observedRecord = *oldLeaderElectionRecord
 		le.observedTime = le.clock.Now()
 	}
-	if le.observedTime.Add(le.config.LeaseDuration).After(now.Time) &&
+	if len(oldLeaderElectionRecord.HolderIdentity) > 0 &&
```

Comment:
What are the compatibility implications here?

Reply:
I'll add a test to prove behavior for this, but older clients watching observe the lease as still held (they don't refresh).  So they assume the holder still holds.  If a new client comes up (N+1) while old clients (N) are still running, the new client would have to acquire the lease, then step down, and the old clients would simply be less aggressive in acquiring.